### ISSUE TITLE
Frame name datetime64 is not JSON serializable

### ISSUE THREAD

**Issue by fhenslee** [static-frame/static-frame#720]
**Description**

When a frame name is numpy datetime64, the error "TypeError: Object of type datetime64 is not JSON serializable" is produced when writing to zip npz. 

The "label_encoder" arg of sf.StoreConfig is not applied to the frame name.

Either "label_encoder" needs to be applied to the frame name, OR, a "frame_name_encorder" arg needs to be added to the sf.StoreConfig?

**Example**

```python
import numpy as np
import static_frame as sf

frame = sf.Frame(
    data=np.random.normal(size=(2, 2)),
    columns=sf.IndexAutoFactory,
    index=sf.IndexAutoFactory,
    name=np.datetime64('2000-01-01'),
)

bus = sf.Bus.from_frames(
    frames=(frame,),
    index_constructor=sf.IndexDate,
)

config = sf.StoreConfig(
    label_encoder=str,
    label_decoder=np.datetime64,
)

fp = r"C:\Users\henslee\Downloads\np_dt_test.npz.zip"
bus.to_zip_npz(fp=fp, config=config)
```

**Platform**

Run the following function (static-frame >= 0.8.1) and provide the results to define your platform and environment:

```python
sf.Platform.display()
Out[3]: 
<Series: platform>
<Index>
platform           Windows-10-10.0.19041-SP0
sys.version        3.8.3 | packaged by conda-forge | (default, Jun  1 2020, 16:59:10) [MSC v.1916 64 bit (AMD64)]
static-frame       0.9.19
numpy              1.19.5
pandas             0.25.3
xlsxwriter         1.1.2
openpyxl           3.0.9
xarray             <ModuleNotFoundError>
tables             3.6.1
pyarrow            0.17.0
msgpack            <ModuleNotFoundError>
msgpack_numpy      <ModuleNotFoundError>
<<U13>             <object>
```

**Comment by flexatone** [comment]
Many thinks for this post. I will give some thought to your suggestions and see what we can do.

Relates to #452

**Comment by fhenslee** [comment]
Thanks. Didn't see the other issue. Sorry for the duplication.

**Comment by flexatone** [comment]
No problem, @fhenslee ! It is not actually a duplicate, as your context is a `Bus` (which offers a clear solution using a `StoreConfig`)... the question is what is the solution for both situations!

**Comment by flexatone** [comment]
Support for both `np.datetime64` and `datetime.date` objects in NPY, NPZ, and npz.zip archives is now implemented and will be in the 1.2 release. Thanks again for this suggestion.

```
>>> f = ff.parse('s(2,2)').rename(np.datetime64('1542-12'))
>>> f.to_npz('/tmp/f.npz')
>>> sf.Frame.from_npz('/tmp/f.npz').name
numpy.datetime64('1542-12')
```

**Comment by fhenslee** [comment]
Thank you @flexatone. Just to confirm, nothing needs to be done with a sf.StoreConfig?
